Hearthstone Deck Tracker/Hearthstone/BattlegroundsBoardState.cs

```diff
@@ -34,8 +35,9 @@ public void SnapshotCurrentBoard()
 				.Where(x => x.IsMinion && x.IsInZone(HearthDb.Enums.Zone.PLAY) && x.IsControlledBy(_game.Opponent.Id))
 				.Select(x => x.Clone())
 				.ToArray();
-			Log.Info($"Snapshotting board state for {opponentHero.Card.Name} with {entities.Length} entities");
-			LastKnownBoardState[GetBattlegroundsBoardState(opponentHero.CardId)] = new BoardSnapshot(entities, _game.GetTurnNumber());
+			Log.Info($"Snapshotting board state for {opponentHero.Card.Name} with cardid {opponentHero.CardId} with {entities.Length} entities");
+			LastKnownBoardState[opponentHero.CardId != null ? GetCorrectBoardstateHeroId(opponentHero.CardId) : opponentHero.CardId] = new BoardSnapshot(entities, _game.GetTurnNumber());
+			Log.Info($"{GetCorrectBoardstateHeroId(opponentHero.CardId)}");
```

Comment:
This can be a little bit cleaner:
```suggestion
			var correctedHero = opponentHero.CardId != null ? GetCorrectBoardstateHeroId(opponentHero.CardId): opponentHero.CardId;
			Log.Info($"Snapshotting board state for {opponentHero.Card.Name} with cardid {opponentHero.CardId} (corrected={correctedHero}) with {entities.Length} entities");
			LastKnownBoardState[correctedHero] = new BoardSnapshot(entities, _game.GetTurnNumber());
```
(I wrote this in the browser, so you should check that it still works correctly.